lib/Lemplate.pm

```diff
@@ -538,6 +534,39 @@ local function stash_set(stash, k, v, default)
     end
 end
 
+_M.vmethods = {
+    join = function(list, delim)
+        if delim == nil then
+            delim = ' '
+        end
+        local out = {}
+        for i = 1, #list, 1 do
+            table.insert(out, list[i])
+            if i ~= #list then
+                table.insert(out, delim)
+            end
+        end
+        return table.concat(out)
+    end,
+
+    first = function(list)
+        return list[1]
+    end,
+
+    push = function(list, elem)
+        table.insert(list, elem)
```

Comment:
Just a side note: we can avoid this `O(n)` operation by caching the table length ourself inside the table, just like how we are handling perl 6 arrays in the fanlang compiler. But we can do such optimizations in `edgelang-fan` instead of here.